app/javascript/components/cloud-volume-form/attach-detach-cloud-volume-form.jsx

```diff
@@ -0,0 +1,89 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+import MiqFormRenderer from '@@ddf';
+import createSchema from './attach-detach-cloud-volume.schema';
+import miqRedirectBack from '../../helpers/miq-redirect-back';
+
+const AttachDetachCloudVolumeForm = ({ recordId, storageManagerId, isAttach, vmChoices }) => {
+                                                // TODO is storageManagerId neeeded?
+
+  const [{ isLoading, deviceMountpointRequired }, setState] = useState({ isLoading: true, deviceMountpointRequired: true });
+
+  useEffect(() => {
+    if (isLoading) {
+
+      API.get(`/api/cloud_volumes/${recordId}`).then((initialValues) => {
+        API.options(`/api/cloud_volumes/${recordId}?ems_id=${initialValues.ems_id}`).then((hellow) => {
+          console.log(hellow)
+        });
+      });
+
+      API.get(`/api/cloud_volumes/${recordId}`).then((initialValues) => {
+        console.log(initialValues)
+        setState((state) => ({
+          ...state,
+          isLoading: false,
+          deviceMountpointRequired: (initialValues.type === 'ManageIQ::Providers::Amazon::StorageManager::Ebs') ? true : false,
```

Comment:
@kavyanekkalapu this is probably out of scope of this changeset but if we have provider specific logic like this we'll need to convert the form to DDF and load this logic from the providers

Reply:
@agrare Sure Melody can add these to provider. @MelsHyrule Can you check providers supporting this feature and add schema in provider side? @GilbertCherrie converted multiple forms like this. Please reach out to him for some samples and help. Thanks.